src/Talis/EchoClient/Base.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Talis\EchoClient;
```

Comment:
to reduce the complexity of the classes the bunch of generic functionality was moved into this new class